### PR TITLE
internal/charmstore: better tests for UploadEntity.

### PR DESCRIPTION
**Pull request by rogpeppe**
Test coverage back up to about 90% in internal/charmstore/addentity.go.

### CODE REVIEW

File: internal/charmstore/addentity.go [hunk 1]
```diff
@@ -111,7 +92,16 @@ func (s *Store) addEntityWithArchive(url *router.ResolvedURL, archive interface{
 //	params.ErrEntityIdNotAllowed if the id may not be created.
 //	params.ErrInvalidEntity if the provided blob is invalid.
 func (s *Store) UploadEntity(url *router.ResolvedURL, blob io.Reader, blobHash string, size int64) error {
-	logger.Infof("uploading entity with blob hash %v", blobHash)
+	// Strictly speaking these tests are redundant, because a ResolvedURL should
+	// always be canonical, but check just in case anyway, as this is
+	// final gateway before a potentially invalid url might be stored
+	// in the database.
+	if url.URL.User == "" {
+		return errgo.WithCausef(nil, params.ErrEntityIdNotAllowed, "entity id does not specify user")
```
Comment (mhilton) [inline internal/charmstore/addentity.go:100]: In general we use "charm or bundle" for "entity" in error messages

File: internal/charmstore/addentity_test.go [hunk 2]
```diff
@@ -26,6 +32,9 @@ type AddEntitySuite struct {
 
 var _ = gc.Suite(&AddEntitySuite{})
 
+func (s *AddEntitySuite) TestUploadEntityError(c *gc.C) {
```
Comment (mhilton) [inline internal/charmstore/addentity_test.go]: I don't think we need this test

**Comment by mhilton** [comment]
:+1:

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1501/
Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1502/
Test FAILed.

File: internal/charmstore/addentity_test.go (hunk 2 above)
Comment (rogpeppe) [inline internal/charmstore/addentity_test.go]: How did you work that out?

File: internal/charmstore/addentity.go (hunk 1 above)
Comment (rogpeppe) [inline internal/charmstore/addentity.go:100]: actually most error messages use "entity". maybe we should standardise on "charm or bundle" but I don't like it as it means we'll need to change all the error messages if/when we add another entity type.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1503/
Test PASSed.

**Comment by jrwren** [comment]
:+1:

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**